GameServerLib/GameObjects/AttackableUnits/AttackableUnit.cs

```diff
@@ -265,6 +295,23 @@ public void SetIsTargetableToTeam(TeamId team, bool targetable)
                 }
             }
         }
+
+        public void ApplyShield(float amount, bool isPhysical, bool isMagical, bool noFade)
```

Comment:
I suggest that instead of booleans and many variables for shields (as I'm sure that there are more than 2), we can use a simple Enum: `PHYSICAL`,`MAGICAL` etc...

By using a simple Enum we can solve most of these boolean checks and wrong arguments.

Reply:
sounds good, and it will simplify those conditions and it's complexity

Reply:
Done, Added Enums type to selecting it's shieldType.